android/app/src/test/java/poke/rogue/helper/presentation/ability/detail/AbilityDetailViewModelTest.kt

```diff
@@ -0,0 +1,59 @@
+package poke.rogue.helper.presentation.ability.detail
+
+import io.kotest.matchers.shouldBe
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import poke.rogue.helper.data.repository.AbilityRepository
+import poke.rogue.helper.presentation.ability.model.toUi
+import poke.rogue.helper.testing.CoroutinesTestExtension
+import poke.rogue.helper.testing.data.repository.FakeAbilityRepository
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@ExtendWith(CoroutinesTestExtension::class)
+class AbilityDetailViewModelTest {
+    private lateinit var repository: AbilityRepository
+    private lateinit var viewModel: AbilityDetailViewModel
+
+    @BeforeEach
+    fun setUp() {
+        repository = FakeAbilityRepository()
+        viewModel = AbilityDetailViewModel(repository)
+    }
+
+    @Test
+    fun `특성 id값으로 특성 상세 정보를 불러온다`() =
+        runTest {
+            // given
+            val abilityId = 1L
+
+            // when
+            viewModel.updateAbilityDetail(abilityId)
+            val abilityDetail = viewModel.abilityDetail.first()
+
+            // then
+            val detail = (abilityDetail as AbilityDetailUiState.Success).data.toUi()
+            detail.title shouldBe "악취"
+            detail.description shouldBe "악취를 풍겨서 공격했을 때 상대가 풀죽을 때가 있다."
+        }
+
+    @Test
+    fun `포켓몬 id 값으로 포켓몬 상세 화면으로 이동한다`() =
+        runTest {
+            // given
+            val pokemonId = 1L
+
+            // when
+            launch {
+                viewModel.navigateToPokemonDetail(pokemonId)
+            }
```

Comment:
```suggestion
viewModel.navigateToPokemonDetail(pokemonId)
```

launch 는 없어도 될 것 같네요

Reply:
navigateToPokemonDetail(pokemonId) 에서 sharedFlow의 값을 다루고 있습니다

launch를 제거하면 
```
kotlinx.coroutines.test.UncompletedCoroutinesError: After waiting for 10s, the test coroutine is not completing
```
코루틴이 끝나지 않는다는 에러가 발생합니다..
찾아보니 launch를 이용해서 테스트하라고 그랬었는데 lauch 없이 테스트하는 방법이 있나요 ?!

Reply:
[꼬상이 참고했다는 글](https://roomedia.tistory.com/entry/Turbine-%EC%97%86%EC%9D%B4-kotest%EC%97%90%EC%84%9C-StateFlow-SharedFlow-%ED%85%8C%EC%8A%A4%ED%8A%B8-%ED%95%98%EA%B8%B0-Unit-Test) 입니다.

Reply:
오둥둥 퀴즈를 참고해주세요!

Reply:
<img width="432" alt="스크린샷 2024-08-02 오후 3 01 48" src="https://github.com/user-attachments/assets/5019e0f4-ce13-4eb7-9c23-7ed38c7b9a7a">